pkg/cluster/cluster_operations_int_test.go

```diff
@@ -77,4 +77,44 @@ var _ = Describe("Basic cluster operations", func() {
 
 	})
 
+	Context("config map creation", func() {
+
+		var objectCleaner *envtestutil.Cleaner
+
+		BeforeEach(func() {
+			objectCleaner = envtestutil.CreateCleaner(envTestClient, envTest.Config, timeout, interval)
+		})
+
+		It("should create configmap with labels and owner reference", func() {
+			// given
+			data := map[string]string{
+				"test-key": "test-value",
+			}
+
+			// when
+			configMap, err := cluster.CreateConfigMap(
+				envTestClient,
+				"config-regs",
+				"default",
+				data,
+				cluster.WithLabels("apps.kubernetes.io/part-of", "opendatahub"),
```

Comment:
I like the DSL bit of it but could we incorporate the same type of functions into the Feature.Manifest processing for instance, so that we have 1 API here not two depending on what you're doing?

```go
feature.Create()..
  .Metadata(WithLabels(), WithOwnerRef())
  .Manifests()..  
```

Reply:
of course